api/v1beta1/azuremachinetemplate_webhook.go

```diff
@@ -62,32 +62,32 @@ func (r *AzureMachineTemplate) ValidateCreate(ctx context.Context, obj runtime.O
 
 	if spec.RoleAssignmentName != "" {
 		allErrs = append(allErrs,
-			field.Invalid(field.NewPath("AzureMachineTemplate", "spec", "template", "spec", "roleAssignmentName"), t, AzureMachineTemplateRoleAssignmentNameMsg),
```

Comment:
I think these changes to `AzureMachineTemplate` aren't needed as they aren't part of the path and just specify the resource type.